File: testdata/jennies/rawtypes/field_with_struct_with_defaults/PHPRawTypes/src/Defaults/DefaultsStructComplexField.php
<?php

namespace Grafana\Foundation\Defaults;

class DefaultsStructComplexField implements \JsonSerializable
{
    public string $uid;

    public \Grafana\Foundation\Defaults\DefaultsStructComplexFieldNested $nested;

    /**
     * @var array<string>
     */
    public array $array;

    /**
     * @param string|null $uid
     * @param \Grafana\Foundation\Defaults\DefaultsStructComplexFieldNested|null $nested
     * @param array<string>|null $array
     */
    public function __construct(?string $uid = null, ?\Grafana\Foundation\Defaults\DefaultsStructComplexFieldNested $nested = null, ?array $array = null)
    {
        $this->uid = $uid ?: "";
        $this->nested = $nested ?: new \Grafana\Foundation\Defaults\DefaultsStructComplexFieldNested();
        $this->array = $array ?: [];
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{uid?: string, nested?: mixed, array?: array<string>} $inputData */
        $data = $inputData;
        return new self(
            uid: $data["uid"] ?? null,
            nested: isset($data["nested"]) ? (function($input) {
    	/** @var array{nestedVal?: string} */
    $val = $input;
    	return \Grafana\Foundation\Defaults\DefaultsStructComplexFieldNested::fromArray($val);
    })($data["nested"]) : null,
            array: $data["array"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->uid = $this->uid;
        $data->nested = $this->nested;
        $data->array = $this->array;
        return $data;
    }
}

File: testdata/jennies/rawtypes/field_with_struct_with_defaults/CSharpRawTypes/src/Grafana/Foundation/Defaults/DefaultsStructComplexField.cs
namespace Grafana.Foundation.Defaults;

using System.Collections.Generic;

public class DefaultsStructComplexField
{
    public string Uid;
    public DefaultsStructComplexFieldNested Nested;
    public List<string> Array;

    public DefaultsStructComplexField()
    {
        this.Uid = "";
        this.Nested = new DefaultsStructComplexFieldNested();
        this.Array = new List<string>();
    }

    public DefaultsStructComplexField(string uid, DefaultsStructComplexFieldNested nested, List<string> array)
    {
        this.Uid = uid;
        this.Nested = nested;
        this.Array = array;
    }
}
